packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1312,6 +1315,42 @@ export function catalogServiceBuilder(
         newEservice
       );
 
+      await repository.createEvent(event);
+    },
+    async deleteRiskAnalysis(
+      eserviceId: EServiceId,
+      riskAnalysisId: RiskAnalysisId,
+      authData: AuthData
+    ): Promise<void> {
+      logger.info(
+        `Deleting Risk Analysis ${riskAnalysisId} for EService ${eserviceId}`
+      );
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      assertRequesterAllowed(eservice.data.producerId, authData.organizationId);
+      assertIsDraftEservice(eservice.data);
+      assertIsReceiveEservice(eservice.data);
+
+      const riskAnalysisToDelete = eservice.data.riskAnalysis.find(
```

Comment:
In the PR where I add the risk analysis update, I created a `retrieveRiskAnalysis` method following the pattern we use also for other entities: https://github.com/pagopa/interop-be-monorepo/pull/254/files#diff-77a5bad1f3b877e7bfc968a9b423c16649fca358d54286d743fe34fa555564ceR151

It retrieves the risk analysis or throws `riskAnalysisNotFound` - maybe you could create it here so that I can reuse it in my PR? Since this will be merged before

Reply:
Updated